packages/beacon-node/src/chain/chain.ts

```diff
@@ -948,6 +952,37 @@ export class BeaconChain implements IBeaconChain {
     if (headState) {
       this.opPool.pruneAll(headState, finalizedState);
     }
+
+    const cpEpoch = cp.epoch;
+    const eip6110Epoch = headState?.config.EIP6110_FORK_EPOCH ?? Infinity;
+
+    if (headState === null) {
+      this.logger.verbose("Head state is null");
+    } else if (cpEpoch >= eip6110Epoch) {
+      const headEpoch = headState.epochCtx.epoch;
+
+      // Set pivotValidatorIndex to 0 if `historicalValidatorLengths` does not
+      // contain validator length for cpEpoch to ensure `newFinalizedValidators` to be empty
+      const pivotValidatorIndex = headState.epochCtx.historicalValidatorLengths.get((headEpoch - cpEpoch + 1) * -1) ?? 0;
```

Comment:
Why is this multiplied by `-1`?

Defaulting to 0 causes unnecessary loop below, why not do an

```
if pivotValidatorIndex != null:
  prune
```

Reply:
Defaulting to 0 causes unnecessary loop below, why not do an

```
if pivotValidatorIndex != null:
  prune
```